src/macros.hpp

```diff
@@ -25,6 +25,7 @@
 #define STRVAL_EMPTY "[[EMPTY]]"
 
 #define WORKSPACE_INVALID -1L
+constexpr int WORKSPACE_NOT_CHANGED = -42;
```

Comment:
uhh no thats not a good idea for an index. Also, `#define` like others.

I'd suggest something like -101

Reasoning:
Regular workspaces start at 1 and grow up
Named start at -1337 and grow down
Special start at -99 and grow up

Reply:
Thanks! Fixed it.
About defines, I was guided by this rule when I did this
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-macros2

Reply:
I realize but we use them there and for the sake of consistency we should keep doing so.